sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ImmutableStorageWithVersioningTest.groovy

```diff
@@ -130,36 +138,37 @@ class ImmutableStorageWithVersioningTest extends APISpec {
                 .endpoint(env.versionedAccount.blobEndpoint)
                 .buildClient()
 
-            def options = new ListBlobContainersOptions().setPrefix(namer.getResourcePrefix())
-            for (def container : cleanupClient.listBlobContainers(options, null)) {
-                def containerClient = cleanupClient.getBlobContainerClient(container.getName())
+            def containerClient = cleanupClient.getBlobContainerClient(vlwContainerName)
+            def containerProperties = containerClient.getProperties()
 
-                if (container.getProperties().getLeaseState() == LeaseStateType.LEASED) {
-                    createLeaseClient(containerClient).breakLeaseWithResponse(new BlobBreakLeaseOptions().setBreakPeriod(Duration.ofSeconds(0)), null, null)
-                }
-                if (container.getProperties().isImmutableStorageWithVersioningEnabled()) {
-                    options = new ListBlobsOptions().setDetails(new BlobListDetails().setRetrieveImmutabilityPolicy(true).setRetrieveLegalHold(true))
-                    for (def blob: containerClient.listBlobs(options, null)) {
-                        def blobClient = containerClient.getBlobClient(blob.getName())
-                        if (blob.getProperties().hasLegalHold()) {
-                            blobClient.setLegalHold(false)
-                        }
-                        if (blob.getProperties().getImmutabilityPolicy().getPolicyMode() != null) {
-                            sleepIfRecord(5 * 1000)
-                            blobClient.deleteImmutabilityPolicy()
-                        }
-                        blobClient.delete()
+            if (containerProperties.getLeaseState() == LeaseStateType.LEASED) {
+                createLeaseClient(containerClient).breakLeaseWithResponse(new BlobBreakLeaseOptions().setBreakPeriod(Duration.ofSeconds(0)), null, null)
+            }
+            if (containerProperties.isImmutableStorageWithVersioningEnabled()) {
+                def options = new ListBlobsOptions().setDetails(new BlobListDetails().setRetrieveImmutabilityPolicy(true).setRetrieveLegalHold(true))
+                for (def blob: containerClient.listBlobs(options, null)) {
```

Comment:
Do the VLW blobs no longer need to be deleted per test?

Reply:
The container is now shared. so we keep piling them up and cleanupSpec now takes care of them.

all these deletions are mainly to make container deletion possible (otherwise it isn't with this feature enabled...)

Reply:
The downside is that now we have to prefix search when we list blobs in few tests, but these tests are not testing listing functionality... so that's fine.

Reply:
I think that is a good pattern to use anyways as it'll be more resilient to any hiccups in the build-up/tear-down process

Reply:
Yup, we had to revisit bunch of tests and prefix search there when we unleashed parallelization.